assets/app/scripts/controllers/attachPersistentVolumeClaim.js

```diff
@@ -0,0 +1,280 @@
+                rebuildBreadcrumb();
+              },
+              function(e) {
+                sendError("The deployment could not be loaded.", getErrorDetails(e));
+              }
+            );
+          }
+
+          DataService.list("persistentvolumeclaims", context, 
+            function(pvcs) {
+              $scope.pvcs = orderByDisplayName(pvcs.by("metadata.name"));
+              if (!$scope.pvcs.length) {
+                sendError("There are no persistent volumes in your project to attach.");
+              } else {
+                if (!$scope.attach.persistentVolumeClaim) {
+                  $scope.attach.persistentVolumeClaim = $scope.pvcs[0];
+                }
+              }
+            },
+            function(e) {
+              sendError("The persistent volume claims could not be loaded.", getErrorDetails(e));
+            }
+          );
+
+          return true;
+        };
+
+        var volumeNameIsUnique = function(name, podTemplate) {
+          if (podTemplate.spec.volumes) {
+            for (var i = 0; i < podTemplate.spec.volumes.length; i++) {
+              var volume = podTemplate.spec.volumes[i];
+              if (volume.name === name) {
+                return false;
+              }
+            }
+          }
+          return true;
+        };
+
+        var volumeMountPathIsUnique = function(name, mountPath, podTemplate) {
+          if (podTemplate.spec.containers) {
+            for (var i = 0; i < podTemplate.spec.containers.length; i++) {
+              var container = podTemplate.spec.containers[i];
+              if ($scope.attach.containers.all || $scope.attach.containers.individual[container.name]) {
+                if (container.volumeMounts) {
+                  for (var j = 0; j < container.volumeMounts.length; j++) {
+                    var volumeMount = container.volumeMounts[j];
+                    if (volumeMount.mountPath === mountPath && name !== volumeMount.Name) {
+                      return false;
+                    }
+                  }
+                }
+              }
+            }          
+          }
+          return true;
+        };
+
+        // breadcrumb must react depending on what we are attaching to (deployment or dc)
+        var rebuildBreadcrumb = function() {
+          $scope.breadcrumbs.splice(1, 0, {
+            title: "Deployments",
+            link: "project/" + $routeParams.project + "/browse/deployments"
+          });
+
+          var deploymentConfig = $scope.attach.deploymentConfig;
+          if (deploymentConfig) {
+            $scope.breadcrumbs.splice(2, 0, {
+              title: deploymentConfig.metadata.name,
+              link: "project/" + $routeParams.project + "/browse/deployments/" + deploymentConfig.metadata.name
+            });
+          }
+
+          var deployment = $scope.attach.deployment;
+          if (deployment) {
+            var deploymentVersion = $filter("annotation")(deployment, "deploymentVersion");
```

Comment:
can't assume an RC is a deployment, so deploymentVersion may not be there, even for an actual deployment (has dep config annotation) we can't assume the deploymentVersion annotation will always be there since anyone can remove annotations